### PR TITLE
Ocpbugs24361 nm state

### PR DESCRIPTION
**Pull request by stevsmit**
Issue appears to date back to 4.12: https://issues.redhat.com/browse/OCPBUGS-18411. https://access.redhat.com/solutions/7031371. 

Unsure how to manage the stacked admonitions. They both belong in that location; one is a tip and the other is important. They do not really belong under the same admonition type. 

Version(s):
4.12+

Issue:
https://issues.redhat.com/browse/OCPBUGS-24361

Link to docs preview:
https://76832--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-example-nmstate-IP-management-dns_k8s_nmstate-updating-node-network-config

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by stevsmit** [comment]
Ayush Garg
  2:56 AM
Hey 
[@Steven Smith](https://redhat-internal.slack.com/team/U01HS5G9JS2)
 o/
[2:56](https://redhat-internal.slack.com/archives/D076NS84F7A/p1717484183819149)
That looks good

File: modules/virt-example-nmstate-IP-management.adoc [hunk 1]
```diff
@@ -117,23 +124,10 @@ dns-resolver:
...
-    ipv6:
-      enabled: true
-      dhcp: true
-      autoconf: true
-      auto-dns: false
 # ...
 ----
 
-* Configure a static DNS for a network interface with a static IP configuration. Note that for this configuration, you must set the `dhcp` parameter to `false` and the `autoconf` parameter to `false`.
+Configure a static DNS for a network interface with an automatic IP configuration. Note that when updating the `/etc/resolv.conf` file on a host node, you must not specify an interface (IPv4 or IPv6) in the `NodeNetworkConfigurationPolicy` (NNCP) manifest.
```
Comment (tmalove) [inline modules/virt-example-nmstate-IP-management.adoc:130]:
```suggestion
* Configure a static DNS for a network interface with an automatic IP configuration. Note that when updating the `/etc/resolv.conf` file on a host node, you must not specify an interface (IPv4 or IPv6) in the `NodeNetworkConfigurationPolicy` (NNCP) manifest.
```

**Review by tmalove — APPROVED** [review]

File: modules/virt-example-nmstate-IP-management.adoc [hunk 2]
```diff
@@ -195,6 +200,54 @@ interfaces:
       auto-dns: true
 # ...
 ----
+<1> If you are using this configuration with dnsmasq or another DNS caching system, consider using `127.0.0.1`.
+
+* Configure a status DNS search option using Dynamic Host Configuration Protocol (DHCP) and IPv6 Stateless Address AutoConfiguration (SLAAC) servers.
```
Comment (qiowang721) [inline modules/virt-example-nmstate-IP-management.adoc:205]: should be 'Configure a static DNS search option using ...'

File: modules/virt-example-nmstate-IP-management.adoc [hunk 3]
```diff
@@ -195,6 +200,54 @@ interfaces:
...
+dns-resolver:
+  config:
+    search:
+    - example.com
+    - example.org
+    server:
+    - 2001:db8:f::1
+    - 192.0.2.251
+# ...
+----
```
Comment (qiowang721) [inline modules/virt-example-nmstate-IP-management.adoc:250]: At the beginning of the DNS example section, it said 'By default, the nmstate API stores DNS values globally...', and the example in Line232-250 is for the default situation, so how about move this example to the beginning of DNS section?

File: modules/virt-example-nmstate-IP-management.adoc (hunk 2 above)
Comment (cathay4t) [reply modules/virt-example-nmstate-IP-management.adoc:205]: I am wondering whether it is supported or not in OpenShift for using `dnsmasq` on `127.0.0.1`.

File: modules/virt-example-nmstate-IP-management.adoc [hunk 4]
```diff
@@ -195,6 +200,54 @@ interfaces:
...
+      enabled: true
+      dhcp: true
+    ipv6:
+      enabled: true
+      dhcp: true
+      autoconf: true
+# ...
+----
+
+* Configure a static DNS for a network interface with an automatic IP configuration. Note that when updating the `/etc/resolv.conf` file on a host node, you must not specify an interface (IPv4 or IPv6) in the `NodeNetworkConfigurationPolicy` (NNCP) manifest.
```
Comment (cathay4t) [inline modules/virt-example-nmstate-IP-management.adoc:229]: If user are not supposed to include any interface when changing DNS in OCP, why above use cases matters?

If openshift has no intention of support `appending static DNS before auto DNS`, then including simple YAML blow is enough.

File: modules/virt-example-nmstate-IP-management.adoc (hunk 3 above)
Comment (stevsmit) [reply modules/virt-example-nmstate-IP-management.adoc]: Moved to beginning.

File: modules/virt-example-nmstate-IP-management.adoc (hunk 4 above)
Comment (stevsmit) [reply modules/virt-example-nmstate-IP-management.adoc]: Only included .Static DNS for network interface with automatic IP configuration

File: modules/virt-example-nmstate-IP-management.adoc (hunk 2 above)
Comment (stevsmit) [reply modules/virt-example-nmstate-IP-management.adoc]: Removed this option and only included .Static DNS for network interface with automatic IP configuration

File: modules/virt-example-nmstate-IP-management.adoc [hunk 5]
```diff
@@ -105,7 +105,30 @@ The following examples show situations that require configuring a network interf
 You cannot use `br-ex` bridge, an OVNKubernetes-managed Open vSwitch bridge, as the interface when configuring DNS resolvers.
 ====
 
-* Configure a static DNS for a network interface with an automatic IP configuration. Note that for this configuration, you must set the `auto-dns` parameter to `false`, so that the Kubernetes NMState Operator can store custom DNS settings for the network interface.
+* Configure a static DNS for a network interface with an automatic IP configuration. Note that when updating the `/etc/resolv.conf` file on a host node, you must not specify an interface (IPv4 or IPv6) in the `NodeNetworkConfigurationPolicy` (NNCP) manifest.
```
Comment (cybertron) [inline modules/virt-example-nmstate-IP-management.adoc:108]: > you must not specify an interface

This isn't strictly true. You _can_ specify an interface, you just don't need to anymore. I think we should soften this language to something like "you do not need to specify an interface".

Comment (stevsmit) [reply modules/virt-example-nmstate-IP-management.adoc:108]: Fixed.

**Comment by qiowang721** [comment]
@stevsmit Thanks for your update, look good to me.
/lgtm

**Comment by aldiazRH** [comment]
LGTM!

**Pull request merged**